pkg/hooks/manager.go

```diff
@@ -252,83 +257,169 @@ func (m *hookManager) onDefineDomainCallback(callback *callBackClient, domainSpe
 			return nil, err
 		}
 		domainSpecXML = result.GetDomainXML()
+	case hooksV1alpha3.Version:
+		client := hooksV1alpha3.NewCallbacksClient(conn)
+		result, err := client.OnDefineDomain(ctx, &hooksV1alpha3.OnDefineDomainParams{
+			DomainXML: domainSpecXML,
+			Vmi:       vmiJSON,
+		})
+		if err != nil {
+			log.Log.Reason(err).Error("Failed to call OnDefineDomain")
+			return nil, err
+		}
+		domainSpecXML = result.GetDomainXML()
 	default:
 		log.Log.Errorf("Unsupported callback version: %s", callback.Version)
 	}
 
 	return domainSpecXML, nil
 }
 
+func preCloudInitIsoDataToJSON(vmi *v1.VirtualMachineInstance, cloudInitData *cloudinit.CloudInitData) ([3][]byte, error) {
```

Comment:
Can't you return individual values (or ptrs to them) instead of the byte slice?

Reply:
Yes, we can do that... I'll change it.